xc-tool-ts/src/config.ts

```diff
@@ -0,0 +1,157 @@
+import { PythUpgradable, PythUpgradable__factory } from "./evm/bindings/index";
+
+import * as ethers from "ethers";
+import * as solWeb3 from "@solana/web3.js";
+import * as borsh from "borsh";
+
+export interface Config {
+  mainnet: MainnetConfig;
+  testnet: TestnetConfig;
+  localDevnet: LocalDevnetConfig;
+}
+
+export interface NetworkConfig {
+  getEvmConfigs(): EvmConfig[];
+  getSolConfigs(): SolConfig[];
+}
+
+export class MainnetConfig implements NetworkConfig {
+  getEvmConfigs(): EvmConfig[] {
+    return [];
+  }
+
+  getSolConfigs(): SolConfig[] {
+    return [];
+  }
+}
+
+export class LocalDevnetConfig implements NetworkConfig {
+  ethereum: EvmConfig;
+  solana: SolConfig;
+  constructor(arg: { ethereum: EvmConfig; solana: SolConfig }) {
+    this.ethereum = arg.ethereum;
+    this.solana = arg.solana;
+  }
+
+  getEvmConfigs(): EvmConfig[] {
+    return [this.ethereum];
+  }
+
+  getSolConfigs(): SolConfig[] {
+    return [this.solana];
+  }
+}
+
+export class TestnetConfig implements NetworkConfig {
+  ethereum: EvmConfig;
+  aurora: EvmConfig;
+  bnb: EvmConfig;
+  solana: SolConfig;
+  pythtest: SolConfig;
+  constructor(
+    arg: {
+      ethereum: EvmConfig;
+      aurora: EvmConfig;
+      bnb: EvmConfig;
+      solana: SolConfig;
+      pythtest: SolConfig;
+    },
+  ) {
+    this.ethereum = arg.ethereum;
+    this.aurora = arg.aurora;
+    this.bnb = arg.bnb;
+    this.solana = arg.solana;
+    this.pythtest = arg.pythtest;
+  }
+
+  getEvmConfigs(): EvmConfig[] {
+    return [this.ethereum, this.aurora, this.bnb];
+  }
+
+  getSolConfigs(): SolConfig[] {
+    return [this.solana, this.pythtest];
+  }
+}
+
+export interface EvmConfig {
+  rpcUrl: string;
+  targetChainContract: string;
+}
+
+export async function getEvmDataSources(cfg: EvmConfig): Promise<any[]> {
+  let provider = new ethers.providers.JsonRpcProvider(cfg.rpcUrl);
+
+  let signer = new ethers.VoidSigner(cfg.targetChainContract, provider);
+
+  let factory = new PythUpgradable__factory(signer);
+
+  let contract = factory.attach(cfg.targetChainContract);
+
+  return await contract.validDataSources();
+}
+
+export interface SolConfig {
+  rpcUrl: string;
+  attesterContract: solWeb3.PublicKey;
+}
+
+export interface AttesterConfigState {
+  owner: solWeb3.PublicKey;
+  whProg: solWeb3.PublicKey;
+  pythOwner: solWeb3.PublicKey;
+  maxBatchSize: number;
+  isActive: boolean;
+  opsOwner: solWeb3.PublicKey | null;
+}
+
+// Helper catch-all object to contain the deserialized schema of SOL accounts
+class Assignable {
```

Comment:
i'm confused about why you need this. maybe explain more in the comment?

Reply:
The Borsh node package is very weird and frustrating. It desperately wants the caller to put a type in first argument of `deserialize`, while also taking a hand-crafted schema object. It does not seem to care about the schema of that aforementioned type. This is mostly copied from Solana's cookbook guide for the JS APIs. I w-was j-just f-following orders.